x-pack/plugins/cases/public/client/helpers/apm/use_cases_transactions.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Transaction } from '@elastic/apm-rum';
+import { useCallback } from 'react';
+import { CommentType } from '../../../../common/api/cases/comment';
+import { CaseAttachmentsWithoutOwner } from '../../../types';
+import { useStartTransaction } from './use_start_transaction';
+
+const CREATE_CASE = 'createCase' as const;
+const ADD_ATTACHMENT_TO_NEW_CASE = 'addAttachmentToNewCase' as const;
+const BULK_ADD_ATTACHMENT_TO_NEW_CASE = 'bulkAddAttachmentsToNewCase' as const;
+const ADD_ATTACHMENT_TO_EXISTING_CASE = 'addAttachmentToExistingCase' as const;
+const BULK_ADD_ATTACHMENT_TO_EXISTING_CASE = 'bulkAddAttachmentsToExistingCase' as const;
+
+export type StartCreateCaseWithAttachmentsTransaction = (param: {
+  appId: string;
+  attachments?: CaseAttachmentsWithoutOwner;
+}) => Transaction | undefined;
+
+// Called when a case is created, attachments are optional
+export const useCreateCaseWithAttachmentsTransaction = () => {
+  const { startTransaction } = useStartTransaction();
+
+  const startCreateCaseWithAttachmentsTransaction =
+    useCallback<StartCreateCaseWithAttachmentsTransaction>(
+      ({ appId, attachments }) => {
+        if (!attachments) {
+          return startTransaction(`Cases [${appId}] ${CREATE_CASE}`);
+        }
+        const alertCount = getAlertCount(attachments);
+        if (alertCount <= 1) {
+          return startTransaction(`Cases [${appId}] ${ADD_ATTACHMENT_TO_NEW_CASE}`);
+        }
+
+        const transaction = startTransaction(`Cases [${appId}] ${BULK_ADD_ATTACHMENT_TO_NEW_CASE}`);
+        transaction?.addLabels({ alert_count: alertCount });
+        return transaction;
+      },
+      [startTransaction]
+    );
+
+  return { startTransaction: startCreateCaseWithAttachmentsTransaction };
+};
+
+export type StartAddAttachmentToExistingCaseTransaction = (param: {
+  appId: string;
+  attachments: CaseAttachmentsWithoutOwner;
+}) => Transaction | undefined;
+
+// Called when attachments are added to existing case
+export const useAddAttachmentToExistingCaseTransaction = () => {
+  const { startTransaction } = useStartTransaction();
+
+  const startAddAttachmentToExistingCaseTransaction =
+    useCallback<StartAddAttachmentToExistingCaseTransaction>(
+      ({ appId, attachments }) => {
+        const alertCount = getAlertCount(attachments);
```

Comment:
What if the `alertCount` is zero? Should we do the same as in `useCreateCaseWithAttachmentsTransaction`?

Reply:
The _Create_ hook can be called with `attachments: undefined` when we are only creating a case, with no attachments. This one is never called with no attachments because the case is already created, it is called when we are attaching something to it.
It is possible though for the `alertCount` to be zero, which means the attachments passed are not alerts, but some other type. In this case, we use the same `ADD_ATTACHMENT_TO_EXISTING_CASE`, since only alerts can be bulk added.
Does it makes sense?

Reply:
Make sense. Thanks, Sergi!